### DISCUSSION TITLE
关于elasticsearch容器搭建建议

### DISCUSSION THREAD

**Discussion by ciscochao** [Bywalks/DarkAngel#17] [General]
使用作者提供的命令启动ES容器时，容易报以下错误：
`ERROR: Elasticsearch did not exit normally - check the logs at /usr/share/elasticsearch/logs/docker-cluster.log
{"type": "server", "timestamp": "2023-03-06T07:23:02,198Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "d4e204b0ae28", "message": "stopping ..." }
{"type": "server", "timestamp": "2023-03-06T07:23:02,218Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "d4e204b0ae28", "message": "stopped" }
{"type": "server", "timestamp": "2023-03-06T07:23:02,218Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "d4e204b0ae28", "message": "closing ..." }`
所以建议在启动的时候指定ES为单节点而不是集群运行可能会更适合小白，下面是单节点启动命令：
`docker run -e ES_JAVA_OPTS="-Xms1024m -Xms1024m" -d -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" --name elasticsearch elasticsearch:7.9.3`

**Comment by Bywalks** [comment]
个人笔记本记录的是单节点启动，一直没发现这个问题，感谢提醒，已采纳更新，加入项目感谢页面 [THANKS.md](https://github.com/Bywalks/DarkAngel/blob/main/THANKS.md)
